sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -492,12 +492,43 @@ object RemoveRedundantAliases extends Rule[LogicalPlan] {
  * Remove no-op operators from the query plan that do not make any modifications.
  */
 object RemoveNoopOperators extends Rule[LogicalPlan] {
+  private def removeAliasOnlyProject(plan: LogicalPlan): LogicalPlan = plan match {
```

Comment:
I am wondering why this is needed, given we already have `Eliminate no-op Projects` below at line 511. Shouldn't this kind of alias-only project already be removed by that?

Reply:
Also just FYI there's some ongoing change for this same rule - https://github.com/apache/spark/pull/31538 ,

Reply:
Alias-only project is not no-op project here.